comp/collector/collector/collectorimpl/agent_check_metadata.go

```diff
@@ -55,12 +55,18 @@ func (p *Payload) SplitPayload(_ int) ([]marshaler.AbstractMarshaler, error) {
 
 // GetPayload builds a payload of all the agentchecks metadata
 func (c *collectorImpl) GetPayload(ctx context.Context) *Payload {
-	hostname, _ := hostname.Get(ctx)
+	hostnameData, _ := hostname.Get(ctx)
 
 	meta := hostMetadataUtils.GetMetaFromCache(ctx, c.config)
-	meta.Hostname = hostname
+	meta.Hostname = hostnameData
 
-	cp := hostMetadataUtils.GetCommonPayload(hostname, c.config)
+	legacyResolutionHostnameData, _ := hostname.GetWithProviderWithoutIMDSV2(ctx)
+	if legacyResolutionHostnameData.Hostname != meta.Hostname {
+		meta.LegacyResolutionHostname = legacyResolutionHostnameData.Hostname
+		meta.HostMetaVersion = "1.0"
```

Comment:
I think this is due to backend implementation, nonetheless I'll let @ogaca-dd answer to this in order to have a precise answer

Reply:
> I think we should have a number for all cases.
I'd probably agree with that unless there's a good reason not to do that

> why not fill LegacyResolutionHostname in all cases even if it's the same as Hostname ?

@hush-hush It's an optimization so that we don't send a useless field to the backend.